httpbakery/client.go

```diff
@@ -485,23 +485,31 @@ func postFormJSON(url string, vals url.Values, resp interface{}, postForm func(u
 	return nil
 }
 
-// RequestMacaroons returns any collections of macaroons from the cookies
-// found in the request. By convention, each slice will contain a primary
-// macaroon followed by its discharges.
+// MacaroonsHeader holds the name of the header that can optionally include
+// a macaroon slice.
+const MacaroonsHeader = "Macaroons"
+
+// RequestMacaroons returns any collections of macaroons from the header and
+// cookies found in the request. By convention, each slice will contain a
+// primary macaroon followed by its discharges.
 func RequestMacaroons(req *http.Request) []macaroon.Slice {
 	var mss []macaroon.Slice
+	h := req.Header.Get(MacaroonsHeader)
```

Comment:
Because it can be done trivially, I think we should allow multiple
instances of the header; e.g.

    for _, h := range req.Header[MacaroonsHeader] {
        etc
    }